Adafruit_DAP_SAM.cpp

```diff
@@ -128,13 +130,101 @@ bool Adafruit_DAP_SAM::select(uint32_t *found_id) {
     if (device->dsu_did == DAP_DSU_did) {
       target_device = *device;
 
+      locked = dap_read_word(DAP_DSU_CTRL_STATUS) & 0x00010000;
+      if (locked) {
+        Serial.println("Device is locked, must be unlocked!");
+      }
+      else {
+        // Stop the core
+        finishReset();
+      }
       return true;
     }
   }
 
   return false;
 }
 
+void Adafruit_DAP_SAM::finishReset() {
+	// Stop the core
+	dap_write_word(DHCSR, 0xa05f0003);
+	dap_write_word(DEMCR, 0x00000001);
+	dap_write_word(AIRCR, 0x05fa0004);
+}
+
+void Adafruit_DAP_SAM::programFlash(const uint8_t * source, uint16_t bootloaderSize, bool doVerify, uint32_t flashOffset) {
+  size_t bufSize = pageSize();
+  uint8_t verifyBuf[bufSize];
+
+  Serial.println("\nProgramming...");
+  uint32_t startAddr = program_start(flashOffset);
+  uint32_t addr = startAddr;
+  while (addr < bootloaderSize) {
+    Serial.print(" 0x");
+    Serial.print(addr, HEX);
+    programBlock(addr, source + addr);
```

Comment:
since this is DAP_SAM, default pagesize for programBlock is 256, seem like it pass the 256 instead of using default of SAMx5 when that instance is called https://github.com/adafruit/Adafruit_DAP/pull/16/files#r559627837